sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5322,8 +5345,11 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
     return Res;
 
   ze_event_handle_t ZeEvent = nullptr;
+  pi_event InternalEvent;
+  bool IsInternal = OutEvent == nullptr;
+  pi_event *Event = OutEvent ? OutEvent : &InternalEvent;
   pi_result Res = createEventAndAssociateQueue(
-      Queue, Event, PI_COMMAND_TYPE_NDRANGE_KERNEL, CommandList);
+      Queue, Event, PI_COMMAND_TYPE_NDRANGE_KERNEL, CommandList, IsInternal);
```

Comment:
what is this pattern (used in many places) aimed to do? (besides obvious setting of the new "Internal" property of the pi_event)

it looks dangerous that you are using an unallocated (dangling) pointer to "InternalEvent"

Reply:
I see that "Event" is actually an output parameter so it is will be allocated inside the call, OK then